src/test/java/io/strimzi/kafka/bridge/http/HttpBridgeTest.java

```diff
@@ -2249,53 +2196,57 @@ public void seekToBeginningAndReceive(TestContext context) {
-                    seekAsync.complete();
+                    context.verify(() -> assertTrue(ar.succeeded()));
+                    context.verify(() -> assertEquals(HttpResponseStatus.NO_CONTENT.code(), ar.result().statusCode()));
+                    seek.complete(true);
                 });
 
-        seekAsync.await();
+        seek.get(60, TimeUnit.SECONDS);
 
         // consume records
-        Async consumeSeekAsync = context.async();
-
         client.get(BRIDGE_PORT, BRIDGE_HOST, baseUri + "/records")
                 .putHeader("Accept", BridgeContentType.KAFKA_JSON_BINARY)
                 .as(BodyCodec.jsonArray())
                 .send(ar -> {
-                    context.assertTrue(ar.succeeded());
+                    context.verify(() -> assertTrue(ar.succeeded()));
 
                     JsonArray body = ar.result().body();
-                    context.assertEquals(10, body.size());
-                    consumeSeekAsync.complete();
+                    context.verify(() -> assertEquals(10, body.size()));
+                    consumeSeek.complete(true);
                 });
 
-        consumeSeekAsync.await();
+        consumeSeek.get(60, TimeUnit.SECONDS);
 
         // consumer deletion
-        Async deleteAsync = context.async();
-
         client.delete(BRIDGE_PORT, BRIDGE_HOST, baseUri)
                 .putHeader("Content-Type", BridgeContentType.KAFKA_JSON)
                 .as(BodyCodec.jsonObject())
                 .send(ar -> {
-                    context.assertTrue(ar.succeeded());
+                    context.verify(() -> assertTrue(ar.succeeded()));
 
                     HttpResponse<JsonObject> response = ar.result();
-                    context.assertEquals(HttpResponseStatus.NO_CONTENT.code(), response.statusCode());
+                    context.verify(() -> assertEquals(HttpResponseStatus.NO_CONTENT.code(), response.statusCode()));
 
-                    deleteAsync.complete();
+                    delete.complete(true);
                 });
 
-        deleteAsync.await();
+        delete.get(60, TimeUnit.SECONDS);
+        context.completeNow();
     }
 
     @Test
-    public void seekToEndAndReceive(TestContext context) {
+    void seekToEndAndReceive(VertxTestContext context) throws InterruptedException, ExecutionException, TimeoutException {
         String topic = "seekToEndAndReceive";
         kafkaCluster.createTopic(topic, 1, 1);
 
-        Async creationAsync = context.async();
+        CompletableFuture<Boolean> produce = new CompletableFuture<>();
+        CompletableFuture<Boolean> create = new CompletableFuture<>();
+        CompletableFuture<Boolean> subscribe = new CompletableFuture<>();
+        CompletableFuture<Boolean> delete = new CompletableFuture<>();
+        CompletableFuture<Boolean> seek = new CompletableFuture<>();
+        CompletableFuture<Boolean> dummy = new CompletableFuture<>();
+        CompletableFuture<Boolean> consumeSeek = new CompletableFuture<>();
```

Comment:
can we create them every time each one is needed?

Reply:
Sure, I was thinking about that as well.